crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -5,17 +5,37 @@ use ra_syntax::{
     ast, AstToken, NodeOrToken, Parse, SmolStr, SyntaxKind, SyntaxKind::*, SyntaxNode,
     SyntaxTreeBuilder, TextRange, TextUnit, T,
 };
+use rustc_hash::FxHashMap;
 use std::iter::successors;
 use tt::buffer::{Cursor, TokenBuffer};
 
 use crate::subtree_source::SubtreeTokenSource;
 use crate::ExpandError;
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum TokenTextRange {
+    Token(TextRange),
+    Delimiter(TextRange, TextRange),
+}
+
+impl TokenTextRange {
+    pub fn range(self, kind: SyntaxKind) -> Option<TextRange> {
```

Comment:
@matklad Do you think I should add this function? Or change `range_by_token` to : 

```rust
fn range_by_token(&self, token_id: tt::TokenId, kind: SyntaxKind) -> TextRange
```

OTOH, I don't know the name of this function, but I don't have a better idea...

Reply:
Rename it to `by_kind`